src/log-store/src/error.rs

```diff
@@ -108,6 +108,16 @@ pub enum Error {
         error: rskafka::client::error::Error,
     },
 
+    #[snafu(display("Failed to resolve Kafka broker endpoint {:?}", broker_endpoint))]
+    ResolveKafkaEndpoint {
+        broker_endpoint: String,
+        #[snafu(source)]
+        error: std::io::Error,
+    },
+
+    #[snafu(display("Failed to find ip for broker endpoint: {:?}", broker_endpoint))]
+    EndpointIpNotFound { broker_endpoint: String },
```

Comment:
We can safely assume that only IPv4 addresses are valid. You can leave a comment to indicate that such an assumption is made.
Thus the `EndpointIpNotFound` error is not necessary and could be removed.

Reply:
Actually, `EndpointIpNotFound` is a workaround, because:
- `lookup_host` return `io::Result<>`, thus `ResolveKafkaEndpoinSnafu` impl `IntoError<Source = Error>
- `find` return `Option<...>`, thus `find.with_context(...)` need a type impl `IntoError<Source = NoneError>`. However it's not possible for `ResolveKafkaEndpointSnafu` to impl `IntoError` again with different `Source`.

Reply:
We can perform `map_err` to use our own composed error rather than the provided error. However, the current codes are ok and I will mark this as resolved.